dts/bindings/audio/nxp,dmic.yaml

```diff
@@ -32,12 +32,12 @@ child-binding:
   description: |
     NXP DMIC channel. Can be used to configure filtering and gain attributes
     of each channel
-  include: base.yaml
-  compatible: "nxp,dmic-channel"
   properties:
     reg:
+      type: array
       required: true
     dmas:
+      type: phandle-array
       required: true
```

Comment:
@mmahadevan108 it looks like the below would be an option. Would that work better for you?
But as I could literally find only **one** occurrence of a child-binding inheriting from base in-tree (so I was wrong earlier, sorry about that), I am really not convinced this is the way to go (https://github.com/zephyrproject-rtos/zephyr/blob/e7db0f8aff81436ff07f557a5ca8e2daa1789188/dts/bindings/mtd/nordic%2Cowned-partitions.yaml#L75-L88)

```
child-binding:
  description: |
    NXP DMIC channel. Can be used to configure filtering and gain attributes
    of each channel

  include:
    name: base.yaml
    property-blocklist:
      - compatible

  properties:
    reg:
      required: true
    dmas:
      required: true
```

Reply:
Oh actually there's also https://github.com/zephyrproject-rtos/zephyr/blob/e7db0f8aff81436ff07f557a5ca8e2daa1789188/dts/bindings/stepper/adi/adi%2Ctmc5041.yaml#L91-L95
but this looks bogus -- it's not "blocking" the compatible property so effectively something like this https://github.com/zephyrproject-rtos/zephyr/blob/e7db0f8aff81436ff07f557a5ca8e2daa1789188/tests/drivers/build_all/stepper/spi.dtsi#L17-L21 is invalid as it misses the required compatible property (and rightfully so, since we don't want one in the first place) (cc @jilaypandya )

Reply:
Do you wanto to have it like this? https://github.com/zephyrproject-rtos/zephyr/pull/68774#discussion_r1689291248